### ISSUE TITLE
Passing array to type-hinted parameter of int doesn't throw an error

### ISSUE THREAD

**Issue by calvinbaart** [peachpiecompiler/peachpie#490]
Sample:
```php
function test(int $a)
{
    return $a;
}

test([]);
```

Peachpie Output:
```
```

PHP output:
```
Fatal error: Uncaught TypeError: Argument 1 passed to test() must be of the type int, array given, called in /Users/calvinbaart/Development/laravel-peachpie-sample/Laravel.Tests/main.php on line 43 and defined in /Users/calvinbaart/Development/laravel-peachpie-sample/Laravel.Tests/main.php:38
Stack trace:
#0 /Users/calvinbaart/Development/laravel-peachpie-sample/Laravel.Tests/main.php(43): test(Array)
#1 {main}
  thrown in /Users/calvinbaart/Development/laravel-peachpie-sample/Laravel.Tests/main.php on line 38
```

A few unit tests for Laravel actually check the exception handling by doing this, so they all fail because this code doesn't throw an exception.

**Comment by jakubmisek** [comment]
Implicit conversion to integer (long) is strict now - it throws `TypeError` or compilation may fail in case the type is known in compile-time.

A diagnostic will be added instead of crashing the compiler in a future update.

**Comment by calvinbaart** [comment]
@jakubmisek Seems that the following code doesn't compile anymore now:
```php
function testFunc2(int $var): int
{
    return $var;
}

function testFunc1(&$var)
{
    return testFunc2($var);
}

$var = 10;
echo testFunc1($var) . "\r\n";
```

The following exception is produced by the compiler
```
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error : Conversion from 'PhpAlias' to 'Int64' not implemented at /mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/main.php(43, 4) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitConvert(TypeSymbol from, TypeRefMask fromHint, TypeSymbol to) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitConvert(BoundExpression expr, TypeSymbol to) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitCall(ILOpCode code, MethodSymbol method, BoundExpression thisExpr, ImmutableArray`1 arguments, BoundTypeRef staticType) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.Semantics.BoundRoutineCall.EmitDirectCall(CodeGenerator cg, ILOpCode opcode, MethodSymbol method, BoundTypeRef staticType) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitConvert(BoundExpression expr, TypeSymbol to) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.Semantics.BoundReturnStatement.Emit(CodeGenerator cg) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at System.Collections.Generic.List`1.ForEach(Action`1 action) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.Semantics.Graph.BoundBlock.Emit(CodeGenerator cg) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.Semantics.Graph.StartBlock.Emit(CodeGenerator cg) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.CodeGen.CodeGenerator.GenerateScope(BoundBlock block, ScopeType type, Int32 to) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.CodeGen.MethodGenerator.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol routine, Action`1 builder, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, Boolean emittingPdb) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at Pchp.CodeAnalysis.SourceCompiler.EmitMethodBody(SourceRoutineSymbol routine) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at System.Threading.Tasks.Parallel.<>c__DisplayClass44_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error : --- End of stack trace from previous location where exception was thrown --- [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at System.Threading.Tasks.Parallel.<>c__DisplayClass44_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int32 timeout, Boolean& replicationDelegateYieldedBeforeCompletion) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at System.Threading.Tasks.TaskReplicator.Replica`1.ExecuteAction(Boolean& yieldedBeforeCompletion) [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]
/home/calvin/.nuget/packages/peachpie.net.sdk/0.9.9-dev/build/Peachpie.NET.Core.Sdk.targets(190,5): error :    at System.Threading.Tasks.TaskReplicator.Replica.Execute() [/mnt/c/Development/laravel-peachpie-sample/Laravel.Tests/Laravel.Tests.msbuildproj]

The build failed. Please fix the build errors and run again.
```
